lib/internal/util/comparisons.js

```diff
@@ -2,6 +2,7 @@
 
 const { compare } = process.binding('buffer');
 const { isArrayBufferView } = require('internal/util/types');
+const { uncurryThis } = require('internal/util/uncurry');
```

Comment:
The `test/parallel/test-modules-bootstrap.js` test will likely need an update... but I'm really wondering if this needs a separate internal module. Why can't this go into `internal/util`?

Reply:
As this function appears three times in the files undef `internal/util` directory, so I think it maybe more concise to be extracted and then simply using by required. On the other hand, as the comment said, it is borrowed from the function with the same name on V8, maybe a bit more efficient to modify and debug when needed in the future ?

And you mean update 76 to 77 in the [test/parallel/test-bootstrap.js](https://github.com/nodejs/node/blob/0227635315c3aa1c31e6814325822a1e4306372e/test/parallel/test-bootstrap-modules.js#L14) ?

Reply:
Yes, that's exactly what is necessary. The test will not fail locally because of how the test is run but we have at least one runner that would fail when adding a new file without increasing that limit in the test.

Reply:
Maybe abstract it a bit `internal/util/functional` or `internal/util/intrinsics` so we can put the copy of `ReflectApply` and probably some more there.
Refs: https://github.com/nodejs/node/pull/12981
Refs: https://github.com/nodejs/node/issues/18795

Reply:
intrinsics needs to happen before NativeModule exists

Reply:
I don't think we should dedicate a file to this function either..`internal/util/functional` sounds good to me, since we also have a `internal/util/types` and it's somewhat in the same category.